src/Commands/LivewireFileManipulationCommandParser.php

```diff
@@ -5,32 +5,45 @@
 use Illuminate\Support\Arr;
 use Illuminate\Support\Str;
 
-class LivewireMakeCommandParser
+class LivewireFileManipulationCommandParser
 {
     protected $appPath;
     protected $viewPath;
     protected $component;
+    protected $newComponent;
     protected $componentClass;
+    protected $newComponentClass;
     protected $directories;
+    protected $newDirectories;
 
-    public function __construct($appPath, $viewPath, $rawCommand)
+    public function __construct($appPath, $viewPath, $rawCommand, $newRawCommand = null)
     {
         $this->appPath = rtrim($appPath, DIRECTORY_SEPARATOR).'/';
         $this->viewPath = rtrim($viewPath, DIRECTORY_SEPARATOR).'/';
 
         $directories = preg_split('/[.]+/', $rawCommand);
+        $newDirectories = preg_split('/[.]+/', $newRawCommand);
 
         $this->component = Str::kebab(array_pop($directories));
         $this->componentClass = Str::studly($this->component);
 
+        $this->newComponent = Str::kebab(array_pop($newDirectories));
+        $this->newComponentClass = Str::studly($this->newComponent);
+
         $this->directories = array_map([Str::class, 'studly'], $directories);
+        $this->newDirectories = array_map([Str::class, 'studly'], $newDirectories);
```

Comment:
Maybe refactor out "newComponent" concept from this class and new up a second parser for the new stuff (extract out the newContent logic bit)

Reply:
I have created a new `LivewireComponentParser` class that deals with all the duplicated name and path getters for the existing and new components. 

The `classContents()` and `newClassContents()` methods remain in the parent `LivewireFileManipulationCommandParser` class because those methods need knowledge of both the existing and new components.